plugins/modules/route53.py

```diff
@@ -806,15 +816,20 @@ def main():
     rr_sets = [camel_dict_to_snake_dict(resource_record_set)]
     formatted_aws = format_record(aws_record, zone_in, zone_id)
     formatted_record = format_record(resource_record_set, zone_in, zone_id)
-
+    if module._diff:
+        module.exit_json(
+            changed=True,
+            wait_id=wait_id,
+            diff=dict(
+                before=formatted_aws,
+                after=camel_dict_to_snake_dict(formatted_record) if command_in != "delete" else {},
+		),
+        	resource_record_sets=[camel_dict_to_snake_dict(formatted_record)] if command_in != "delete" else {},
```

Comment:
this is should not be part of the diff structure

Reply:
to avoid duplicate code, you can do something like

```
result = dict(changed=True, wait_id=wait_id, resource_record_sets=[...])
if module._diff:
    result.update(dict(before=formatted_aws, after=....))
module.exit_json(**result)
```